### ISSUE TITLE
Improvement: Support Logstash (experimental) feature --allow-env

### ISSUE THREAD

**Issue by SchulteDev** [logstash-plugins/logstash-input-jmx#24]
Logstash has the experimental feature [--allow-env](https://www.elastic.co/guide/en/logstash/current/environment-variables.html). This feature enables "using Environment Variables in the Configuration".

For this plugin, logstash-input-jmx, please provide support for this Logstash-feature. In special, jmxonf-files do not respect the feature (plugin configuration file does already).

**Comment by untergeek** [comment]
So, you want Logstash to read these files and interpret environment variables when reading them?

This is not a bad feature, but is not likely to be added by the core Logstash team, due to our high workload maintaining and extending Logstash itself.  

We only directly maintain a set of core plugins, even though we host the repositories for many community supported ones (the chief benefit of which is centralization of documentation).  If you, or someone else in the community wants to add that feature in and submit a PR (preferably with unit and integration tests), we'd be happy to review it and merge it. We welcome [community maintainers](https://www.elastic.co/guide/en/logstash/current/community-maintainer.html) of plugins, as they tend to be passionate about making things work for themselves and others.

**Comment by SchulteDev** [comment]
Hi @untergeek,

this issue is more plugin specific. I would like Logstash to support environment variables, correct, but that's another issue (the both of us are discussing that at [logstash#6081](https://github.com/elastic/logstash/issues/6081)). Somehow, Logstash supports environment variables already by --allow-env.

In this issue, I am requesting the plugin "logstash-input-jmx" to support --allow-env.

For using logstash-input-jmx I have (at least) two configuration files. 
First one is the input-declaration, which is a "normal" Logstash-configuration file, containing "input{jmx{...}}" etc. This one supports --allow-env right now, env-placeholder are understood.
Second configuration file is "logstash-input-jmx" specific, saved at a different location. This file contains stuff like JMX-host, JMX-queries and so on. As far as I can see, --allow-env does not affect those configuration files at all. Placing placeholders like ${TCP_PORT} here seems not to be evaluated.

**Comment by untergeek** [comment]
Indeed, that is what I understood. Thank you for confirming. As stated, I don't see us getting to it soon. If you'd like to see this feature, you or another community member will have to add it and submit a pull request.
